urllib3/poolmanager.py

```diff
@@ -204,22 +212,46 @@ def connection_from_pool_key(self, pool_key):
                 return pool
 
             # Make a fresh ConnectionPool of the desired type
-            pool = self._new_pool(pool_key.scheme, pool_key.host, pool_key.port)
+            pool = self._new_pool(pool_key.scheme, pool_key.host, pool_key.port,
+                                  pool_kwargs=pool_kwargs)
             self.pools[pool_key] = pool
 
         return pool
 
-    def connection_from_url(self, url):
+    def _merge_pool_kwargs(self, override):
         """
-        Similar to :func:`urllib3.connectionpool.connection_from_url` but
-        doesn't pass any additional parameters to the
-        :class:`urllib3.connectionpool.ConnectionPool` constructor.
+        Merge a dictionary of override values for self.connection_pool_kw.
 
-        Additional parameters are taken from the :class:`.PoolManager`
-        constructor.
+        This does not modify self.connection_pool_kw and returns a new dict.
+        Any keys in the override dictionary with a value of ``None`` are
+        removed from the merged dictionary.
+        """
+        base_pool_kwargs = self.connection_pool_kw.copy()
+        if override:
+            for key, value in override.items():
+                if value is None:
+                    try:
+                        del base_pool_kwargs[key]
+                    except KeyError:
+                        pass
+                else:
+                    base_pool_kwargs[key] = value
+        return base_pool_kwargs
```

Comment:
Should this method come *after* `connection_from_url`, just to keep the `connection_from_*` methods in a visual block?